core/src/main/java/org/infinispan/security/actions/SecurityActions.java

```diff
@@ -0,0 +1,187 @@
+      GetCacheComponentRegistryAction action = new GetCacheComponentRegistryAction(advancedCache);
+      return doPrivileged(action);
+   }
+
+   public static void undefineConfiguration(EmbeddedCacheManager cacheManager, String name) {
+      UndefineConfigurationAction action = new UndefineConfigurationAction(cacheManager, name);
+      doPrivileged(action);
+   }
+
+   public static AuthorizationManager getCacheAuthorizationManager(AdvancedCache<?, ?> cache) {
+      return doPrivileged(cache::getAuthorizationManager);
+   }
+
+   public static void addListener(EmbeddedCacheManager cacheManager, Object listener) {
+      doPrivileged(new AddCacheManagerListenerAction(cacheManager, listener));
+   }
+
+   public static CompletionStage<Void> removeListenerAsync(Listenable listenable, Object listener) {
+      RemoveListenerAsyncAction action = new RemoveListenerAsyncAction(listenable, listener);
+      return doPrivileged(action);
+   }
+
+   public static DistributionManager getDistributionManager(final Cache<?, ?> cache) {
+      GetCacheDistributionManagerAction action = new GetCacheDistributionManagerAction(cache.getAdvancedCache());
+      return doPrivileged(action);
+   }
+
+   public static <K, V> CompletionStage<CacheEntry<K, V>> getCacheEntryAsync(final AdvancedCache<K, V> cache, K key) {
+      GetCacheEntryAsyncAction<K, V> action = new GetCacheEntryAsyncAction<>(cache, key);
+      return doPrivileged(action);
+   }
+
+   public static Configuration getCacheConfiguration(final AdvancedCache<?, ?> cache) {
+      GetCacheConfigurationAction action = new GetCacheConfigurationAction(cache);
+      return doPrivileged(action);
+   }
+
+   public static <K> CompletionStage<Boolean> cacheContainsKeyAsync(AdvancedCache<K, ?> ac, K key) {
+      return doPrivileged(new CacheContainsKeyAsyncAction<>(ac, key));
+   }
+
+   public static void addCacheDependency(EmbeddedCacheManager cacheManager, String from, String to) {
+      doPrivileged(new AddCacheDependencyAction(cacheManager, from, to));
+   }
+
+   public static Configuration getCacheConfigurationFromManager(final EmbeddedCacheManager cacheManager, String cacheName) {
+      GetCacheConfigurationFromManagerAction action = new GetCacheConfigurationFromManagerAction(cacheManager, cacheName);
+      return doPrivileged(action);
+   }
+
+   public static PersistenceManager getPersistenceManager(final EmbeddedCacheManager cacheManager, String cacheName) {
+      final GetPersistenceManagerAction action = new GetPersistenceManagerAction(cacheManager, cacheName);
+      return doPrivileged(action);
+   }
+
+   public static Health getHealth(final EmbeddedCacheManager cacheManager) {
+      GetCacheManagerHealthAction action = new GetCacheManagerHealthAction(cacheManager);
+      return doPrivileged(action);
+   }
+
+   public static <K, V> ComponentRegistry getComponentRegistry(AdvancedCache<K, V> cache) {
```

Comment:
can be removed. Duplicates `ComponentRegistry getCacheComponentRegistry(AdvancedCache<?, ?> advancedCache)`